web/packages/teleport/src/services/api/api.ts

```diff
@@ -58,21 +59,38 @@ const api = {
   },
 
   delete(url, data?) {
-    return api.fetchJson(url, {
+    return api.fetchJsonWithMfaAuthnRetry(url, {
       body: JSON.stringify(data),
       method: 'DELETE',
     });
   },
 
   put(url, data) {
-    return api.fetchJson(url, {
+    return api.fetchJsonWithMfaAuthnRetry(url, {
       body: JSON.stringify(data),
       method: 'PUT',
     });
   },
 
-  async fetchJson(url, params): Promise<any> {
-    const response = await api.fetch(url, params);
+  /**
+   * fetchJsonWithMfaAuthnRetry calls on `api.fetch` and
+   * processes the response.
+   *
+   * It returns the JSON data if it is a valid JSON and
+   * there were no response errors.
+   *
+   * If a response had an error and it contained a MFA authn
+   * required message, then a retry is attempted after a user
+   * successfully re-authenticates with an MFA device.
+   *
+   * All other errors will be thrown.
+   */
+  async fetchJsonWithMfaAuthnRetry(
+    url: string,
+    customOptions: RequestInit,
+    webauthnAssertResp?: WebauthnAssertionResponse
```

Comment:
We have both `webauthnAssertResp` and `webauthnResponse` which acts hold the same value but need to be named differently so that they don't override each other.

I think it'd make the code easier to understand if we named this argument `webauthnResponse` (so that it mirrors the argument to `fetch`) and the one defined later before retrying could be named like `webauthnResponseForRetry`.